tests/tools/test_dist_part.py

```diff
@@ -0,0 +1,226 @@
+
+def test_partition():
+    # Step1: graph partition
+    output_dir = '2parts'
+    os.system('python tools/partition_algorithms/random_partition.py '\
+              '--metadata chunked-data/metadata.json --output_path {} --num_partitions 2'.format(output_dir))
+    for ntype in ['author', 'institution', 'paper']:
+        fname = os.path.join(output_dir, '{}.txt'.format(ntype))
+        with open(fname, 'r') as f:
+            header = f.readline().rstrip()
+            assert isinstance(int(header), int)
+
+def test_dispatch():
+    # Step2: data dispatch
+    out_dir = 'partitioned'
+    num_parts = 2
+    with open('ip_config.txt', 'w') as f:
+        f.write('127.0.0.1\n')
+        f.write('127.0.0.2\n')
+
+    os.system('python tools/dispatch_data.py --in-dir chunked-data '\
+              '--partitions-dir 2parts --out-dir {} --ip-config ip_config.txt'.format(out_dir))
+
+    # check metadata.json
+    meta_fname = os.path.join(out_dir, 'metadata.json')
+    with open(meta_fname, 'rb') as f:
+        meta_data = json.load(f)
+
+    all_etypes = ['affiliated_with', 'writes', 'cites', 'rev_writes']
+    for etype in all_etypes:
+        assert len(meta_data['edge_map'][etype]) == num_parts
+    assert meta_data['etypes'].keys() == set(all_etypes)
+    assert meta_data['graph_name'] == 'mag240m'
+
+    all_ntypes = ['author', 'institution', 'paper']
+    for ntype in all_ntypes:
+        assert len(meta_data['node_map'][ntype]) == num_parts
+    assert meta_data['ntypes'].keys() == set(all_ntypes)
+    assert meta_data['num_edges'] == 4200
+    assert meta_data['num_nodes'] == 720
+    assert meta_data['num_parts'] == num_parts
+    assert meta_data['part_method'] == 'metis'
+
+    for i in range(num_parts):
+        sub_dir = 'part-' + str(i)
+        assert meta_data[sub_dir]['node_feats'] == 'part{}/node_feat.dgl'.format(i)
+        assert meta_data[sub_dir]['edge_feats'] == 'part{}/edge_feat.dgl'.format(i)
+        assert meta_data[sub_dir]['part_graph'] == 'part{}/graph.dgl'.format(i)
+
+        # check data
+        sub_dir = os.path.join(out_dir, 'part' + str(i))
+
+        # graph.dgl
+        fname = os.path.join(sub_dir, 'graph.dgl')
+        assert os.path.isfile(fname)
+        g_list, data_dict = load_graphs(fname)
+        g = g_list[0]
+        assert isinstance(g, dgl.DGLGraph)
+
+        # node_feat.dgl
+        fname = os.path.join(sub_dir, 'node_feat.dgl')
+        assert os.path.isfile(fname)
+        tensor_dict = load_tensors(fname)
+        all_tensors = ['paper/feat', 'paper/label', 'paper/year']
+        assert tensor_dict.keys() == set(all_tensors)
+        for key in all_tensors:
+            assert isinstance(tensor_dict[key], torch.Tensor)
+
+        # edge_feat.dgl
+        fname = os.path.join(sub_dir, 'edge_feat.dgl')
+        assert os.path.isfile(fname)
+        tensor_dict = load_tensors(fname)
+
+if __name__ == '__main__':
+    test_chunk_graph()
```

Comment:
One general question: these tests are required to run sequentially and in strict order due to latter ones depend on outputs from former ones? If so, I think we'd better to put them into a single `test_xxx` function to avoid ambiguity. And according to the code, I think these functions are quite coupled such as `dir`, `magic numbers like num_edges`.

Reply:
I'm fine with it. What do you think? @jermainewang

Reply:
pls let me add more. Personally, I assume all the tests could be run independently and I usually run a specific test via `python -m pytest tests/distributed/test_rpc.py::test_xxx`. But for tests here, it's not possible except the first one. So putting them into single `test_xxx` function may avoid such confusion.

Reply:
Got it.

Reply:
You should use pytest fixtures to ensure test orders https://docs.pytest.org/en/7.1.x/how-to/fixtures.html

Reply:
@jermainewang `pytest fixtures` work for `python -m pytest tests/tools/test_dist_part.py`. However it prohibits specific tests like `python -m pytest tests/tools/test_dist_part.py::test_chunk_graph`.

Reply:
Rui's solution is also fine for me.

Reply:
I've changed the three test functions to one test function.